app/controllers/assessments_controller.rb

```diff
@@ -66,9 +66,11 @@ class AssessmentsController < ApplicationController
 
   def index
     @is_instructor = @cud.has_auth_level? :instructor
-    @announcements = Announcement.where("start_date < :now AND end_date > :now", now: Time.now)
-                                 .where("course_id = ? OR system = ?", @course.id, true)
-                                 .where(persistent: false).order(:start_date)
+    announcements_tmp = Announcement.where("start_date < :now AND end_date > :now",
+       now: Time.now)
+      .where(persistent: false)
+    @announcements = announcements_tmp.where(course_id: @course.id)
```

Comment:
Just a nit, but why is it necessary to create a temp variable here?

Reply:
look at the line below. Announcements_tmp is referenced again. The activerecord .or is not very flexible

Reply:
Chaskiel's comment is correct